### DISCUSSION TITLE
Custom Nested Groups

### DISCUSSION THREAD

**Discussion by lukecharle** [tailwindlabs/tailwindcss#9713] [Help] [answered]
So I've come across [Differentiating nested groups](https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups) today and its come in quite useful. 

I've actually setup some custom group variants, for example:

```
addVariant('current', '&.current');
addVariant('group-current', ':merge(.group).current &');
```

My issue now is that I cannot use the nested selector with this, so I would expect to use for example `group/menu` and then `group-current/menu:text-black` but this doesn't work out of the box.

As you can see from the screenshot, `menutrigger`gets added to the `group-current` part, but not at the beginning so it won't work. Do I need to modifiy config and update the variants I've created?

<img width="480" alt="Screenshot 2022-11-02 at 14 13 15" src="https://user-images.githubusercontent.com/115987937/199512041-3a46c8ed-4e93-44a0-bcc3-6a0e53f448a7.png">

**Comment by wongjn** [comment, marked as answer]
There's not really any documentation on this but I was able to adapt the [core implementation](https://github.com/tailwindlabs/tailwindcss/blob/88dcb6ebc51093ab2289dd7405c6c915d863ad71/src/corePlugins.js#L151-L170) to work for [your custom case in this Tailwind Play](https://play.tailwindcss.com/RW1ppVates?file=config) (open the *Generated CSS* panel to see the generated utilities).

**Reply by lukecharle** [reply]
Ah perfect! Would there be a way to implement this in a more dynamic way? So I don't need to go and add values everytime I use a new nested group?

**Reply by wongjn** [reply]
I'm not sure what you mean – the solution will generate the rules with the correct selector for any `group-current/<label>` class usage.
